src/components/document/sorted-documents.tsx

```diff
@@ -1,37 +1,93 @@
 import React, { useState } from "react";
 import { observer } from "mobx-react";
-import { DocumentContextReact } from "./document-context";
 import classNames from "classnames";
-import ArrowIcon from "../../assets/icons/arrow/arrow.svg";
+
+import { DocumentContextReact } from "./document-context";
 import { SortedDocument } from "../../models/stores/sorted-documents";
 import { DocumentModelType, getDocumentContext } from "../../models/document/document";
 import { DecoratedDocumentThumbnailItem } from "../thumbnail/decorated-document-thumbnail-item";
 import { useStores } from "../../hooks/use-stores";
 import { logDocumentViewEvent } from "../../models/document/log-document-event";
 import { ENavTab } from "../../models/view/nav-tabs";
+import { DocFilterType } from "../../models/stores/ui-types";
+import { SimpleDocumentItem } from "../thumbnail/simple-document-item";
+import { IDocumentMetadata } from "../../../functions/src/shared";
+
+import ArrowIcon from "../../assets/icons/arrow/arrow.svg";
 
 import "./sort-work-view.scss";
 
 interface IProps {
+  docFilter: DocFilterType;
   idx: number;
   sortedSection: SortedDocument
 }
 
 export const SortedDocuments: React.FC<IProps> = observer(function SortedDocuments(props: IProps) {
-  const { idx, sortedSection } = props;
-  const { persistentUI } = useStores();
-
+  const { docFilter, idx, sortedSection } = props;
+  const { persistentUI, sortedDocuments } = useStores();
   const [showDocuments, setShowDocuments] = useState(false);
 
-  const handleSelectDocument = (document: DocumentModelType) => {
+  const getDocument = (docKey: string) => {
+    const document = sortedDocuments.documents.all.find((doc: DocumentModelType) => doc.key === docKey);
+    if (document) return document;
+
+    // Calling `fetchFullDocument` will update the `documents` store with the full document,
+    // triggering a re-render of this component since its an observer.
+    sortedDocuments.fetchFullDocument(docKey);
+
+    return undefined;
+  };
+
+  const documentCount = () => {
+    if (docFilter !== "Problem") return sortedSection.documents.length;
```

Comment:
Why is this different for the Problem filter?

Reply:
I think I must've misunderstood why I was seeing incorrect counts for the Problem filter at some point. The other method for counting below seems to work fine for problems now, so I've removed this line.